### PR TITLE
[feat] Improve media panel rendering performance

### PR DESCRIPTION
**Pull request by KaroMourad**
- decrease amount of unnecessarily renderings on mouse hovering(especially outside of the media panel)
- memoize functions (use callback hooks)
- separate `DepthDropdown` and `DepthSlider` components to have more readable and reusable code

### CODE REVIEW

File: aim/web/ui/src/components/MediaSet/MediaSet.tsx [hunk 1]
```diff
@@ -507,6 +383,166 @@ const MediaGroupedList = React.memo(function MediaGroupedList({
 },
 areEqual);
 
+function DepthSlider({
```
Comment (roubkar) [inline aim/web/ui/src/components/MediaSet/MediaSet.tsx:386]: please move this component to a separate file

File: aim/web/ui/src/components/MediaSet/MediaSet.tsx [hunk 2]
```diff
@@ -507,6 +383,166 @@ const MediaGroupedList = React.memo(function MediaGroupedList({
...
+              <Icon name='arrow-right' fontSize={10} />
+            </Button>
+          }
+        />
+      </div>
+    </ErrorBoundary>
+  );
+}
+
+function DepthDropdown({
```
Comment (roubkar) [inline aim/web/ui/src/components/MediaSet/MediaSet.tsx:447]: please move this component to a separate file

**Review by roubkar — CHANGES_REQUESTED** [review]

File: aim/web/ui/src/components/MediaSet/MediaSet.tsx (hunk 1 above)
Comment (KaroMourad) [reply aim/web/ui/src/components/MediaSet/MediaSet.tsx]: done , thanks

File: aim/web/ui/src/components/MediaSet/MediaSet.tsx (hunk 2 above)
Comment (KaroMourad) [reply aim/web/ui/src/components/MediaSet/MediaSet.tsx]: done , thanks

**Review by arsengit — APPROVED** [review]

**Review by Hamik25 — APPROVED** [review]

**Review by roubkar — APPROVED** [review]

**Pull request merged**